src/lib/pages/backoffice/Providers/ProviderDetails/ProviderDetails.js

```diff
@@ -21,61 +22,74 @@ import {
+import { ProviderInformation } from './ProviderInformation';
 
-const DeleteLibraryButton = (props) => {
+const DeleteProviderButton = (props) => {
   return (
     <DeleteButton
       fluid
-      content="Delete library"
+      content="Delete provider"
       labelPosition="left"
       {...props}
     />
   );
 };
 
 class ActionMenu extends React.Component {
-  createRefProps(libraryPid) {
-    const brwReqRefProps = {
-      refType: 'Borrowing Requests',
-      onRefClick: (brwReqPid) =>
-        goTo(ILLRoutes.borrowinRequestDetailsFor(brwReqPid)),
-      getRefData: () =>
-        borrowingRequestApi.list(
-          borrowingRequestApi.query().withLibraryPid(libraryPid).qs()
-        ),
+  createRefProps(providerPid, providerType) {
+    let refType;
+    let path;
+    let query;
+    if (providerType === 'LIBRARY') {
+      refType = 'Borrowing Requests';
+      path = ILLRoutes.borrowinRequestDetailsFor;
+      query = borrowingRequestApi.list(
+        borrowingRequestApi.query().withProviderPid(providerPid).qs()
+      );
+    } else {
+      refType = 'Order';
+      path = AcquisitionRoutes.orderDetailsFor;
+      query = orderApi.list(orderApi.query().withProviderPid(providerPid).qs());
+    }
+    const refProps = {
+      refType: refType,
+      onRefClick: (pid) => goTo(path(pid)),
+      getRefData: () => query,
     };
 
-    return [brwReqRefProps];
+    return [refProps];
   }
 
   render() {
     const {
-      data: { metadata: library },
-      deleteLibraryHandler,
+      data: { metadata: provider },
+      deleteProviderHandler,
     } = this.props;
 
     return (
       <div className="bo-action-menu">
         <Sticky context={this.contextRef}>
-          <EditButton fluid to={ILLRoutes.libraryEditFor(library.pid)} />
+          <EditButton fluid to={ProviderRoutes.providerEditFor(provider.pid)} />
           <DeleteRecordModal
-            deleteHeader={`Are you sure you want to delete the Library record
-              with ID ${library.pid}?`}
-            refProps={this.createRefProps(library.pid)}
-            onDelete={() => deleteLibraryHandler(library.pid)}
-            trigger={DeleteLibraryButton}
+            deleteHeader={`Are you sure you want to delete the Provider record
+              with ID ${provider.pid}?`}
+            refProps={this.createRefProps(provider.pid, provider.type)}
+            onDelete={() => deleteProviderHandler(provider.pid)}
+            trigger={DeleteProviderButton}
           />
 
           <Divider horizontal>Navigation</Divider>
 
           <Link
             to={ILLRoutes.borrowingRequestListWithQuery(
-              borrowingRequestApi.query().withLibraryPid(library.pid).qs()
+              borrowingRequestApi.query().withProviderPid(provider.pid).qs()
             )}
           >
             <Icon name="search" />
-            See borrowing requests from {library.name}
+            See {provider.type === 'LIBRARY'
```

Comment:
Same here, they might be mixing...
So you can put:
`See <link>acquisition orders</link> or <link>ILL borrowing requests</link> from {provider.name}`